sdk/python/flet/control.py

```diff
@@ -50,8 +50,9 @@ def __init__(
         self.__data: Any = None
         self.data = data
         self.__event_handlers = {}
-        if ref:
-            ref.current = self
+        self.__ref = ref
+        if self.__ref:
+            self.__ref.current = self
 
```

Comment:
It's unnecessary to duplicate `ref` setter logic in the constructor. You could just remove:

```
        self.__ref = ref
        if self.__ref:
            self.__ref.current = self
```

and call property setter right away:

```
self.ref = ref
```

So, why would we need to keep the last used reference to a reference inside control: `self.__ref`?

Reply:
Maybe you could provide a sample of how you are going to use that?

Reply:
Let's say you wanted to reuse a component inside your application, but you need to use it with a ref because of your architecture or for some other reason. With this change, you could create the component in a `components/` module and simply import it wherever you connect it to the respective ref. Is that understandable or should I provide an example with code?

Reply:
🤯 Please! Example wouldn't hurt! :)

Reply:
https://github.com/iron3oxide/flet-mvp-counter

Since the example had to be a bit more complex I simply modified my MVP counter example. You can find the most relevant sections in view.py and the components module.

Reply:
I just realized that I could also pass the ref to the function/method that returns the control and use it there. If there's anything that would make adding the ref property as proposed here impractical, I'd be fine doing that instead.

Reply:
If it works for you then I'd better close this PR. Honestly, I don't see a point of keeping "last used" reference inside a control instance.